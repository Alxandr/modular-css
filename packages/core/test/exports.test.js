"use strict";

const dedent = require("dedent");
const namer  = require("test-utils/namer.js");
const { cwd, find } = require("test-utils/fixtures.js");

const Processor = require("../processor.js");

describe("/processor.js", () => {
    describe("exports", () => {
        var processor;
        
        beforeEach(() => {
            processor = new Processor({
                cwd,
                namer,
            });
        });
        
        it("should export an object of arrays containing strings", async () => {
            const result = await processor.string(
                "./simple.css",
                dedent(`
                    .red { color: red; }
                    .black { background: #000; }
                    .one, .two { composes: red, black; }
                `)
            );

            expect(result.exports).toMatchSnapshot();
        });

        it("should export identifiers and their classes", async () => {
            await processor.file(find("start.css"));
            
            const output = await processor.output();
            
            expect(output.compositions).toMatchSnapshot();
        });
    });
});
