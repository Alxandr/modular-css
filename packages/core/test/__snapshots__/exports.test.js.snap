// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/processor.js exports should export an object of arrays containing strings 1`] = `
Object {
  "black": Array [
    "black",
  ],
  "one": Array [
    "red",
    "black",
    "one",
  ],
  "red": Array [
    "red",
  ],
  "two": Array [
    "red",
    "black",
    "two",
  ],
}
`;

exports[`/processor.js exports should export identifiers and their classes 1`] = `
Object {
  "fixtures/folder/folder.css": Object {
    "folder": "folder",
  },
  "fixtures/local.css": Object {
    "booga": "booga",
    "folder": "white",
    "looga": "booga looga",
    "one": "red",
    "two": "blue",
  },
  "fixtures/start.css": Object {
    "booga": "booga",
    "folder": "white",
    "one": "red",
    "tooga": "tooga",
    "two": "blue",
    "wooga": "booga wooga",
  },
}
`;
